### ISSUE TITLE
[BUG] Doesn't work with Proton on Linux

### ISSUE THREAD

**Issue by javier-posse** [TopazTK/KH-ReFined#59]
I now it may be out of scope for the project, but the original games do work with Proton 7-9, and the mod just crash with any version.
Using SteamOS 3.2 on SteamDeck, oppened with the "--language es" param.

If someone could look into it would be awesome, but understandable if ignored.

**Comment by KHOmega** [comment]
> I have been able to make it work on Lutris after installing dotnet48 through winetricks I can't test with Proton (doesn't work for me)
> 
> I have noticed that the game can't be closed anymore through normal means (you can only kill the process) It also messes up fullscreen, needing to select borderless then press Alt + Tab to work correctly

Are you able to walk me on how to install dotnet48 through winetricks on my Steam Deck? This would help with beta testing new ReFixed features on the platform.

**Comment by Frousties** [comment]
I don't know much about the Steam Deck, but you can try to install and use prontontricks.
To get the appid of a non-steam game, you can look into ```~/.steam/root/userdata/[your account id]/760/screenshots.vdf``` and find the name, it will have the appid on its left.

**Comment by KHOmega** [comment]
> I don't know much about the Steam Deck, but you can try to install and use prontontricks. To get the appid of a non-steam game, you can look into `~/.steam/root/userdata/[your account id]/760/screenshots.vdf` and find the name, it will have the appid on its left.

I have dotnet48 installed, and ReFixed will not launch via Lutris. SteamOS might have some other things pertaining to how it handles some of these things, so for now, this method does not work on Steam Deck. Hopefully something is figured out soon.

**Comment by The-Math-God** [comment]
any luck with this yet?

**Comment by bunktech** [comment]
> I have been able to make it work on Lutris after installing dotnet48 through winetricks I can't test with Proton (doesn't work for me)
> 
> I have noticed that the game can't be closed anymore through normal means (you can only kill the process) It also messes up fullscreen, needing to select borderless then press Alt + Tab to work correctly

Which wine version were you using in lutris?

**Comment by Frousties** [comment]
> > I have been able to make it work on Lutris after installing dotnet48 through winetricks I can't test with Proton (doesn't work for me)
> > I have noticed that the game can't be closed anymore through normal means (you can only kill the process) It also messes up fullscreen, needing to select borderless then press Alt + Tab to work correctly
> 
> Which wine version were you using in lutris?

Lutris 6.4

**Comment by bunktech** [comment]
@Frousties Thanks! I was able to get it working on the Steam Deck with that version + dotnet in winetricks!

@KHOmega If you're using lutris, you can set up a new game with the Lutris 6.4 runner that @Frousties mentioned and execute the game installer in the wineprefix to get it installed. When you select the game in the main Lutris UI, you'll see the Launch and Wine options at the bottom of the UI, if you expand out the arrow next to the Wine glass, you can select "Winetricks". It will open the winetricks UI, where the first thing I would suggest is to Disable Silent Install (you'll see it at the bottom of the list), then the dialog will remain open, you can select the default wineprefix option (it's the default option), and install a windows component dll in the next dialog. From there you'll be able to install from a selection of apps. Scroll down to dotnet48, check it, and proceed to install. You'll get a bunch of warnings and errors about 64-bit vs 32-bit prefix, don't worry about it, and also an error about dotnet not being supported on wine 6.4 -- this doesn't seem to affect anything. Proceed through all the installers as it will progressively install from dotnet 40 to dotnet 48. Once this is complete, you should be able to launch with the KH Re:Fined exe. 

Note - I only tested KH Final Mix. Haven't tested the others, but I imagine they'll all work at this point.

**Comment by KHOmega** [comment]
> @Frousties Thanks! I was able to get it working on the Steam Deck with that version + dotnet in winetricks!
> 
> @KHOmega If you're using lutris, you can set up a new game with the Lutris 6.4 runner that @Frousties mentioned and execute the game installer in the wineprefix to get it installed. When you select the game in the main Lutris UI, you'll see the Launch and Wine options at the bottom of the UI, if you expand out the arrow next to the Wine glass, you can select "Winetricks". It will open the winetricks UI, where the first thing I would suggest is to Disable Silent Install (you'll see it at the bottom of the list), then the dialog will remain open, you can select the default wineprefix option (it's the default option), and install a windows component dll in the next dialog. From there you'll be able to install from a selection of apps. Scroll down to dotnet48, check it, and proceed to install. You'll get a bunch of warnings and errors about 64-bit vs 32-bit prefix, don't worry about it, and also an error about dotnet not being supported on wine 6.4 -- this doesn't seem to affect anything. Proceed through all the installers as it will progressively install from dotnet 40 to dotnet 48. Once this is complete, you should be able to launch with the KH Re:Fined exe.
> 
> Note - I only tested KH Final Mix. Haven't tested the others, but I imagine they'll all work at this point.

This is the solution! All four public current releases (Re:Fined 3.0) do indeed work with this method on Steam Deck.

**Comment by TopazTK** [comment]
Closing, will put this as a tutorial in the main repository soon!
Thanks to every single one of you for your tremendous work on this.
